.circleci/config.yml

```diff
@@ -350,71 +333,58 @@ jobs:
           command: pnpm -r test:package
 
 workflows:
-  version: 2
   pipeline:
     when:
       equal: [pipeline, << pipeline.parameters.workflow >>]
     jobs:
       - checkout:
           <<: *default-context
+          name: 'Checkout'
       - test_unit:
           <<: *default-context
           name: 'JSDOM tests'
           requires:
-            - checkout
+            - Checkout
```

Comment:
These should probably remain `checkout` as it relies on `key`, not the `name`. 🤔 
The jobs seem stuck now. :/
```suggestion
            - checkout
```

Reply:
+1 to keep snake case to match with the rest of the codebase and with https://circleci.com/docs/introduction-to-yaml-configurations/

Reply:
It does rely on a name. The job seemed stuck as Github checks expect a job with the old name. I will update the GH configuration once the PR is merged so that other PRs aren't stuck.

Reply:
@oliviertassinari I haven't found anything in the CircleCI that forces or recommends any particular format. In fact, they use different conventions in different parts of the docs (https://circleci.com/docs/configuration-reference/#requires has kebab-case)

I prioritized the readability of GH status checks (especially for external contributors).
Compare
![image](https://github.com/user-attachments/assets/93b005eb-5c74-4ea9-8d2c-d16fd651a50e)
with
![image](https://github.com/user-attachments/assets/7cc30ddd-9ec3-41c1-acc9-d0ab8e12576e)

Reply:
I'm all for more readability if it's just a GH config. 👍